scripts/generate_share_images/index.ts

```diff
@@ -10,48 +10,76 @@ import fs from 'fs-extra';
 import path from 'path';
 import Pageres from 'pageres';
 import urlJoin from 'url-join';
-import { STATES } from '../../src/common';
-
-// We don't care about the values here, but this is a cheap way to determine all
-// of the counties we have any data for and should therefore screenshot.
-import CalculatedCountyInterventionJSON from '../../src/assets/data/calculated_county_interventions.json';
-const COUNTIES = Object.keys(CalculatedCountyInterventionJSON);
+import {
+  fetchAllStateProjections,
+  fetchAllCountyProjections,
+} from '../../src/common/utils/model';
+import { Projections } from '../../src/common/models/Projections';
+import { Metric, ALL_METRICS } from '../../src/common/metric';
 
 const BASE_URL = 'http://localhost:3000/internal/share-image';
 const CSS_SELECTOR = '.screenshot';
 const OUTPUT_DIR = path.join(__dirname, 'output');
-const OUTPUT_SIZE = '1200x630';
 // How many screenshots to send to pageres at once.
 const PAGERES_BATCH_SIZE = 10;
 
+const SHARE_OUTPUT_SIZE = '1200x630';
+const EXPORT_OUTPUT_SIZE = '2400x1350';
+
 const BLACKLISTED_COUNTIES = [
   '11001', // DC - We treat it as a state, not a county.
 ];
 
 (async () => {
   await fs.ensureDir(OUTPUT_DIR);
   await fs.emptyDir(OUTPUT_DIR);
-  await fs.emptyDir(`${OUTPUT_DIR}/states`);
-  await fs.emptyDir(`${OUTPUT_DIR}/counties`);
 
-  let screenshots = [] as Array<{ url: string; filename: string }>;
+  console.log('Fetching projections...');
+  const allStatesProjections = await fetchAllStateProjections();
+  const allCountiesProjections = await fetchAllCountyProjections();
+  console.log('Fetch complete.');
 
-  screenshots.push({ url: '/', filename: 'home' });
+  let screenshots = [] as Array<{ url: string; filename: string, outputSize: string }>;
 
-  for (const stateCode in STATES) {
-    const state = stateCode.toLowerCase();
+  // Homepage share image.
+  screenshots.push({ url: '/', filename: 'home', outputSize: SHARE_OUTPUT_SIZE });
+
+  function addScreenshotsForLocation(relativeUrl: string, projections: Projections) {
+    // Overall share image.
     screenshots.push({
-      url: `/states/${state}`,
-      filename: `/states/${state}`,
+      url: relativeUrl,
+      filename: relativeUrl,
+      outputSize: SHARE_OUTPUT_SIZE,
     });
+
+    // Chart images.
+    for (const metric of ALL_METRICS) {
+      if (metric === Metric.FUTURE_PROJECTIONS || projections.getMetricValue(metric) !== null) {
+        const shareUrl = urlJoin(relativeUrl, '/chart/', '' + metric);
+        const exportUrl = urlJoin(shareUrl, '/export');
```

Comment:
Maybe we should unify the URL generation logic at some point to ensure that they are consistent between the index file generation and the share images generation, not critical right now though.

Reply:
Yeah, it's a good idea.  I added a TODO for now.